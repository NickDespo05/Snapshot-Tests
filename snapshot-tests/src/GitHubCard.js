import Card from "react-bootstrap/Card";
import { useEffect, useState } from "react";

export default function GitHubCard() {
    const [gitHubStuff, setGitHubStuff] = useState(null);

    useEffect(() => {
        fetch("https://api.github.com/users/NickDespo05")
            .then((res) => res.json())
            .then((data) => {
                setGitHubStuff(
                    <div>
                        <p>{data.name} </p>
                        <p>{data.bio}</p>
                        <img src={data.avatar_url} />
                    </div>
                );
            });
    }, []);

    return (
        <div>
            <Card>{gitHubStuff}</Card>
        </div>
    );
}
