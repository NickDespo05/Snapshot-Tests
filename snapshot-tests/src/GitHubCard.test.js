import GitHubCard from "./GitHubCard";
import renderer from "react-test-renderer";

test("renders the bio, name, and profile picture to the file", () => {
    const tree = renderer.create(<GitHubCard />).toJSON();
    expect(tree).toMatchSnapshot();
});
