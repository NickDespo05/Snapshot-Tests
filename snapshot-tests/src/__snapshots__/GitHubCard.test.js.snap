// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the bio, name, and profile picture to the file 1`] = `
<div>
  <div
    className="card"
  />
</div>
`;
